js/ui/components/clubhouse.js

```diff
@@ -112,6 +112,148 @@ var ClubhouseNotificationSource = new Lang.Class({
     },
 });
 
+var ClubhouseButtonManager = new Lang.Class({
+    Name: 'ClubhouseButtonManager',
+
+    _init: function() {
+        this.parent();
+
+        this._iconSize = 64
+
+        this._openButton =
+            new St.Button({ name: 'clubhouse-open-button',
+                            child: new St.Icon({ style_class: 'clubhouse-open-button-icon' }),
+                          });
+        this._openButton.connect('clicked', () => {
+            this.emit('open-clubhouse');
+        })
+
+        Main.layoutManager.addChrome(this._openButton);
+
+        this._closeButton =
+            new St.Button({ name: 'clubhouse-close-button',
+                            child: new St.Icon({ style_class: 'clubhouse-close-button-icon' }),
+                          });
+        this._closeButton.connect('clicked', () => {
+            this.emit('close-clubhouse');
+        })
+
+        Main.layoutManager.addChrome(this._closeButton);
+
+        this._clubhouseWindowActor = null;
+        this._trackWindowActor(this._getClubhouseActorFromWM());
+
+        this.updateVisibility();
+        this.reposition();
+
+        this._windowMapHandler =
+            global.window_manager.connect('map', Lang.bind(this, this._mapWindow));
+
+        this._overviewShowingHandler = Main.overview.connect('showing', () => {
+            this.updateVisibility();
+        });
+
+        this._overviewHiddenHandler = Main.overview.connect('hidden', () => {
+            this.updateVisibility();
+        });
+    },
+
+    _getClubhouseActorFromWM: function() {
+        for (let actor of global.get_window_actors()) {
+            let window = actor.meta_window;
+            if (window && window.get_gtk_application_id() == CLUBHOUSE_ID)
+                return actor;
+        }
+
+        return null;
+    },
+
+    _updateCloseButtonPosition: function() {
+        this._closeButton.y = this._openButton.y;
+        if (this._clubhouseWindowActor)
+            this._closeButton.x = this._clubhouseWindowActor.x - this._iconSize / 2;
+    },
+
+    reposition: function() {
+        let monitor = Main.layoutManager.primaryMonitor;
+        if (!monitor)
+            return;
+
+        this._openButton.x = monitor.width - this._iconSize / 2;
```

Comment:
If you use `this._iconSize` for this calculation, you're assuming that the button will not add any borders/margins/paddings that change its size. That may be true at the moment, but a better way to get this information is to look at the width of the button.